packages/mongodb-chatbot-server/src/services/dataStreamer.ts

```diff
@@ -24,7 +24,6 @@ function makeServerSentEventDispatcher<
       res.writeHead(200, {
         "Cache-Control": "no-cache",
         "Content-Type": "text/event-stream",
-        "Access-Control-Allow-Origin": "*",
```

Comment:
needed to remove this to have CORS work. if i didn't remove this, then the browser was showing this CORS err:

```
Access to fetch at 'http://localhost:3000/api/v1/conversations/65bacd2f27c07aaa82302c05/messages?stream=true' 
from origin 'http://localhost:5173' has been blocked by CORS policy: 
The value of the 'Access-Control-Allow-Origin' header in the response 
must not be the wildcard '*' when the request's credentials mode is 'include'.
```

Reply:
As is we're only including credentials in non-prod environments. Will this still work in prod envs where we aren't allowing cookies/credentials?

Reply:
it was when i played around w/ it in local dev, but maybe a solution is to let the headers here be configurable when creating the data streamer. that way we can keep as is for prod and mutate as we wish for staging

Reply:
given we're now allowing credentials (including cookies) per https://github.com/mongodb/chatbot/pull/308#discussion_r1492822479, this shouldn't be a concern

Reply:
> it was when i played around w/ it in local dev, but maybe a solution is to let the headers here be configurable when creating the data streamer. that way we can keep as is for prod and mutate as we wish for staging

i'd rather not add this b/c i don't think it's necessary and i'd rather not add a feature for the sake of adding the feature.